shadows/framework/src/main/java/org/robolectric/shadows/ShadowWebView.java

```diff
@@ -261,36 +288,64 @@ public WebChromeClient getWebChromeClient() {
 
   @Implementation
   public boolean canGoBack() {
-    return canGoBack;
+    // TODO: Remove the canGoBack check when setCanGoBack is deleted.
+    if (canGoBackIsSet) {
+      return canGoBack;
+    }
+    return history.size() > 1;
   }
 
   @Implementation
   public void goBack() {
-    goBackInvocations++;
+    if (canGoBack()) {
+      goBackInvocations++;
+      // TODO: Delete this when setCanGoBack is deleted, since this creates two different behavior
```

Comment:
Could you fix an issue described in this TODO or remove a comment if not actual?